### PR TITLE
Minimal logging support, debug frame dump logging

### LINKED ISSUE #96: Frame logging
rsocket-cli tracking issue https://github.com/rsocket/rsocket-cli/issues/100

Ideally with a simple flag, add ability to see either of

RSocket Frames (requestn, setup, payload) or
Transport Frames (Websocket frame, potentially HTTP2)

The former is way more important to me.

### PR DESCRIPTION
**Pull request by whyoleg**
Provides frame logging (with handwritten hex dump) closes #96

### CODE REVIEW

File: rsocket-core/src/commonMain/kotlin/io/rsocket/kotlin/connection/LoggingConnection.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import io.ktor.utils.io.core.*
+import io.rsocket.kotlin.frame.*
+import kotlinx.coroutines.*
+
+class LoggingConnection(private val delegate: Connection) : Connection {
+    override val job: Job get() = delegate.job
+
+    private fun log(tag: String, packet: ByteReadPacket) {
+        println("\n$tag: ${packet.dumpFrameToString()}")
```
Comment (yschimke) [inline rsocket-core/src/commonMain/kotlin/io/rsocket/kotlin/connection/LoggingConnection.kt:27]: System.err would be my preference or pipe through java.util.logging.  What is the logging API for rsocket-kotlin?

Comment (whyoleg) [reply rsocket-core/src/commonMain/kotlin/io/rsocket/kotlin/connection/LoggingConnection.kt:27]: For now, there is no standard multiplatform logging library exist as I know. 
But I found one, that works everywhere - https://github.com/touchlab/Kermit. Touchlab is now one of the biggest companies, who are creating libraries for multiplatform, so I was thinking about using their lib.

File: rsocket-core/src/commonMain/kotlin/io/rsocket/kotlin/frame/ErrorFrame.kt [hunk 2]
```diff
@@ -34,10 +35,18 @@ class ErrorFrame(
             else -> writePacket(data)
         }
     }
+
+    override fun StringBuilder.appendFlags(): Unit = Unit
+    override fun StringBuilder.appendSelf() {
+        append("\nError code: ").append(errorCode).append("[").append(throwable::class.simpleName).append("]")
+        if (throwable.message != null) append(" Message: ").append(throwable.message)
+        if (data != null) appendPacket("Data:", data)
```
[the remark this answers is not in the document]
Comment (whyoleg) [reply rsocket-core/src/commonMain/kotlin/io/rsocket/kotlin/frame/ErrorFrame.kt:43]: No, under the hood, `appendPacket` will add new line before tag and hex dump of data

**Review by yschimke — APPROVED** [review]

**Comment by yschimke** [comment]
The logging approach looks like a good balance for now

**Pull request merged**